### DISCUSSION TITLE
dhcpcd-11 will be privsep only

### DISCUSSION THREAD

**Discussion by rsmarples** [rsmarples/dhcpcd#279] [Ideas]
The privsep implementation dhcpcd offers has proven robust and has been the default since dhcpcd-10.
To simplify the code path, the option to NOT have privsep will be removed to simplify the code path even more.
For Linux, we still have the` --disable-seccomp` configure option when something in libc changes we haven't codified yet.

This idea is currently blocked by #278 as I do like having IllumOS as a supported OS.

**Comment by jxcang** [comment]
Please see https://github.com/NetworkConfiguration/dhcpcd/issues/87#issuecomment-4446918219

If we could agree on how to improve privsep to be more utilizing `libcap` and `systemd` compatible, I would appreciate that.
Right now, as I have discussed in the above comment, would over-privileging some proxy processes, or over-privileging the privileged process.
